app/scripts/views/mixins/flow-begin-mixin.js

```diff
@@ -7,21 +7,27 @@
 define(function (require, exports, module) {
   'use strict';
 
-  var $ = require('jquery');
-  var AuthErrors = require('lib/auth-errors');
+  var Flow = require('models/flow');
 
   module.exports = {
     afterRender: function () {
-      var flowId = this.user.get('flowId');
-      var flowBeginTime = parseInt($('body').attr('data-flow-begin'), 10);
+      var self = this;
 
-      if (! flowBeginTime) {
-        flowBeginTime = undefined;
-        this.logError(AuthErrors.toError('INVALID_DATA_FLOW_BEGIN_ATTR'));
-      }
+      self.flow = new Flow({
+        sentryMetrics: self.sentryMetrics,
+        window: self.window
+      });
 
-      this.metrics.setActivityEventMetadata('flowBeginTime', flowBeginTime);
-      this.metrics.logFlowBegin(flowId, flowBeginTime);
+      return self.flow.fetch().then(function () {
```

Comment:
This is only returning a promise because of `Flow.fetch`, which doesn't actually need to do anything asynchronous.  Following your advice in https://github.com/mozilla/fxa-content-server/issues/3789 I think the cleanest fix will be to avoid `Flow` having an async `fetch` method at all, then the `afterRender` mixin here doesn't need to return a promise.